nexosim/tests/integration/simulation_halt.rs

```diff
@@ -0,0 +1,91 @@
+//! Simulation pause and resume
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::{Duration, Instant};
+
+use nexosim::model::{Context, InitializedModel, Model};
+use nexosim::ports::{EventQueue, Output};
+use nexosim::simulation::{ExecutionError, Mailbox, SimInit, SimulationError};
+use nexosim::time::{MonotonicTime, SystemClock};
+
+struct RecurringModel {
+    pub message: Output<()>,
+    delay: Duration,
+}
+impl RecurringModel {
+    fn new(delay: Duration) -> Self {
+        Self {
+            delay,
+            message: Output::new(),
+        }
+    }
+    async fn process(&mut self) {
+        self.message.send(()).await;
+    }
+}
+impl Model for RecurringModel {
+    async fn init(self, cx: &mut Context<Self>) -> InitializedModel<Self> {
+        cx.schedule_periodic_event(self.delay, self.delay, RecurringModel::process, ())
+            .unwrap();
+        self.into()
+    }
+}
+
+#[test]
+fn halt_and_resume() -> Result<(), SimulationError> {
+    let mut model = RecurringModel::new(Duration::from_secs(2));
+    let mailbox = Mailbox::new();
+
+    let message = EventQueue::new();
+    model.message.connect_sink(&message);
+    let mut message = message.into_reader();
+
+    let t0 = MonotonicTime::EPOCH;
+
+    let (simu, mut scheduler) = SimInit::new()
+        .add_model(model, mailbox, "timed_model")
+        .set_clock(SystemClock::from_instant(t0, Instant::now()))
+        .init(t0)?;
+
+    let simulation = Arc::new(Mutex::new(simu));
+    let spawned_simulation = simulation.clone();
+
+    thread::spawn(move || spawned_simulation.lock().unwrap().step_unbounded());
```

Comment:
I woul join this thread later, just in case.

Reply:
Joined in the latest commit.